anydex/wallet/iota/iota_wallet.py

```diff
@@ -0,0 +1,334 @@
+import os
+import re
+from abc import ABCMeta
+from asyncio import Future
+
+from iota.crypto.types import Seed
+from iota.transaction import ProposedTransaction
+from iota.types import Address
+from ipv8.util import succeed, fail
+from sqlalchemy import exists
+
+from anydex.wallet.cryptocurrency import Cryptocurrency
+from anydex.wallet.iota.iota_database import initialize_db, \
+    DatabaseSeed, DatabaseTransaction, DatabaseBundle, DatabaseAddress
+from anydex.wallet.iota.iota_provider import IotaProvider
+from anydex.wallet.wallet import Wallet, InsufficientFunds
+
+
+class AbstractIotaWallet(Wallet, metaclass=ABCMeta):
+
+    def __init__(self, db_path, testnet, node):
+        super().__init__()
+
+        self.name = 'Iota Test Network' if testnet else 'Iota Network'
```

Comment:
In the Monero and Ethereum wallet implementations the TESTNET parameter is not passed to the constructor, instead it has a fixed `False` value for a regular wallet, and a `True` value for a testnet wallet. I think we should discuss what approach we should take globally for all wallets.

Reply:
True, although I think we agreed today to follow the bitcoinlib schema and rather have child classes inherit from abstract classes instead of having a testnet version of the class that patches up the concrete class

Reply:
Therefore, we will most likely be lightly adjusting the structure of Ethereum and Monero as part of the refactoring